astropy/io/fits/tests/test_connect.py

```diff
@@ -57,13 +60,35 @@ def equal_data(a, b):
     return all(np.all(a[name] == b[name]) for name in a.dtype.names)
 
 
+@contextlib.contextmanager
+def working_directory(path):
+    """
+    A context manager changing the working directory to the given
+    path. The directory is changed back on exit.
+    """
+
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+
+
 class TestSingleTable:
     def setup_class(self):
         self.data = np.array(
             list(zip([1, 2, 3, 4], ["a", "b", "c", "d"], [2.3, 4.5, 6.7, 8.9])),
             dtype=[("a", int), ("b", "U1"), ("c", float)],
         )
 
+    def test_path(self, tmp_path):
+        with working_directory(tmp_path):
```

Comment:
Like you, I am now very perplexed on why this bug only shows up when the filename has no path attached to it. 🤯 

Perhaps we should wait for @saimn to review. He might have extra understanding in regards to `io.fits`. 🙏

Reply:
ref https://github.com/astropy/astropy/issues/15350#issuecomment-2551993980